images/router/haproxy/conf/haproxy-config.template

```diff
@@ -175,6 +180,10 @@ frontend fe_sni
   bind 127.0.0.1:{{env "ROUTER_SERVICE_SNI_PORT" "10444"}} ssl no-sslv3 {{ if gt (len .DefaultCertificate) 0 }}crt {{.DefaultCertificate}}{{ else }}crt /var/lib/haproxy/conf/default_pub_keys.pem{{ end }} crt-list /var/lib/haproxy/conf/cert_config.map accept-proxy
   mode http
 
+  # DNS labels are case insensitive (RFC 4343), we need to convert the hostname into lowercase
+  # before matching, or any requests containing uppercase characters alawys do not match.
+  http-request set-header Host %[req.hdr(Host),lower]
```

Comment:
Actually... should we be changing this here?  Or should we make all of the matches tolerate a mixed case hostname (i.e. in router.go's generateRouteRegexp).